pkg/resmap/idslice_test.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resmap
+
+import (
+	"testing"
+
+	"github.com/kubernetes-sigs/kustomize/pkg/resource"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestLess(t *testing.T) {
+	ids := IdSlice{
+		resource.NewResId(schema.GroupVersionKind{Kind: "ConfigMap"}, "cm"),
+		resource.NewResId(schema.GroupVersionKind{Kind: "Pod"}, "pod"),
+		resource.NewResId(schema.GroupVersionKind{Kind: "Namespace"}, "ns1"),
+		resource.NewResId(schema.GroupVersionKind{Kind: "Namespace"}, "ns2"),
+	}
+	type index struct{ i, j int }
+
+	indexes := []index{
+		{i: 0, j: 1},
+		{i: 2, j: 0},
+		{i: 3, j: 0},
+		{i: 3, j: 1},
+		{i: 2, j: 3},
+	}
+
+	for _, ind := range indexes {
+		if !ids.Less(ind.i, ind.j) {
+			t.Fatalf("%+v should be in front of %+v", ids[ind.i].Gvk(), ids[ind.j].Gvk())
+		}
+	}
```

Comment:
Another way to test would be invoking `sort.Sort(ids)` and compare the output with the expected ordering using reflect pkg.

Reply:
I'm going to use `sort`. It seems to have better readability. Thank you :)

Reply:
This is update. PTAL